This patch makes three changes to the exim_install script:

*) Changes the test used to determine whether a file should be installed
   to cope with files which we "touch" to zero bytes for packaging
   purposes.

*) Causes the configure file to be installed as a sample only.  The
   administrator needs to rename this file to "activate" exim.

*) Changes the invocation of make-info to be less prone to fail on
   silly errors.  XXX Check that this is still necessary from time
   to time.

--- scripts/exim_install.orig	Mon Aug  2 17:43:03 1999
+++ scripts/exim_install	Tue Aug  3 14:31:20 1999
@@ -146,9 +146,9 @@
     from=../util/
   fi
 
-  if [ ! -s ${from}${name} ]; then
+  if [ ! -r ${from}${name} ]; then
     echo $com ""
-    echo $com "*** `pwd`/${from}${name} does not exist or is empty"
+    echo $com "*** `pwd`/${from}${name} does not exist"
     echo $com "*** Have you built Exim successfully?"
     echo $com "*** Exim installation ${ver}failed ***"
     exit 1
@@ -199,25 +199,20 @@
   fi
 done
 
-
-
-# If there is no configuration file, install the default.
+# Install only a sample configuration, which encourages the admin to look
+# at it, since Exim won't run without a configure file.
 
 echo $com ""
 
-if [ ! -f ${CONFIGURE_FILE} ]; then
-  echo $com Installing default configuration in ${CONFIGURE_FILE}
-  echo $com because there is no existing configuration file.
-  echo ${CP} ../src/configure.default ${CONFIGURE_FILE}
-  ${real} ${CP} ../src/configure.default ${CONFIGURE_FILE}
+  echo $com Installing sample configuration file
+  ${real} mkdir -p ${CONFIGURE_FILE%/*}
+  echo ${CP} ../src/configure.default ${CONFIGURE_FILE}.sample
+  ${real} ${CP} ../src/configure.default ${CONFIGURE_FILE}.sample
   if [ $? -ne 0 ]; then
     echo $com ""
     echo $com "*** Exim installation ${ver}failed ***"
     exit 1
   fi
-else
-  echo $com Configuration file ${CONFIGURE_FILE} already exists
-fi
 
 # Install info files if the directory is defined and the Texinfo
 # source documentation is present.
@@ -239,19 +234,21 @@
   echo $com Info installation directory is ${INFO_DIRECTORY}
   echo $com ""
 
-  makeinfo --no-split --output exim_overview.info ../doc/oview.texinfo
+  makeinfo --no-validate --no-split --output exim_overview.info \
+      ../doc/oview.texinfo
   echo ${CP} exim_overview.info ${INFO_DIRECTORY}
   ${real} ${CP} exim_overview.info ${INFO_DIRECTORY}
   install-info --section="Exim" \
       --entry "* Overview: (exim_overview).    Overview of the Exim system" \
       ${INFO_DIRECTORY}/exim_overview.info ${INFO_DIRECTORY}/dir
-  makeinfo --no-split --output exim.info ../doc/spec.texinfo
+  makeinfo --no-validate --no-split --output exim.info ../doc/spec.texinfo
   echo ${CP} exim.info ${INFO_DIRECTORY}
   ${real} ${CP} exim.info ${INFO_DIRECTORY}
   install-info --section="Exim" \
       --entry "* User guide: (exim).           Exim manual" \
       ${INFO_DIRECTORY}/exim.info ${INFO_DIRECTORY}/dir
-  makeinfo --no-split --output exim_filter.info ../doc/filter.texinfo
+  makeinfo --no-validate --no-split --output exim_filter.info \
+      ../doc/filter.texinfo
   echo ${CP} exim_filter.info ${INFO_DIRECTORY}
   ${real} ${CP} exim_filter.info ${INFO_DIRECTORY}
   install-info --section="Exim" \
